### PR TITLE
enhance: separate serializer logic from sync task 

### PR DESCRIPTION
**Pull request by congqixia**
See also #27675

Since serialization segment buffer does not related to sync manager can
shall be done before submit into sync manager. So that the pk statistic
file could be more accurate and reduce complex logic inside sync
manager.

### CODE REVIEW

File: internal/datanode/syncmgr/task.go [hunk 1]
```diff
@@ -118,30 +140,39 @@ func (t *SyncTask) Run() error {
...
-	if t.insertData != nil {
-		totalSize += float64(t.insertData.GetMemorySize())
-	}
-	metrics.DataNodeFlushedSize.WithLabelValues(fmt.Sprint(paramtable.GetNodeID()), metrics.AllLabel, metricSegLevel).Add(totalSize)
-	metrics.DataNodeEncodeBufferLatency.WithLabelValues(fmt.Sprint(paramtable.GetNodeID()), metricSegLevel).Observe(float64(t.tr.RecordSpan().Milliseconds()))
+		if t.insertData != nil {
+			totalSize += float64(t.insertData.GetMemorySize())
+		}
+		metrics.DataNodeFlushedSize.WithLabelValues(fmt.Sprint(paramtable.GetNodeID()), metrics.AllLabel, metricSegLevel).Add(totalSize)
+		metrics.DataNodeEncodeBufferLatency.WithLabelValues(fmt.Sprint(paramtable.GetNodeID()), metricSegLevel).Observe(float64(t.tr.RecordSpan().Milliseconds()))*/
```
Comment (XuanYang-cn) [inline internal/datanode/syncmgr/task.go:175]: Please add back these metrics

**Review by XuanYang-cn — APPROVED** [review]
/lgtm

File: internal/datanode/syncmgr/task.go (hunk 1 above)
Comment (congqixia) [reply internal/datanode/syncmgr/task.go:175]: done

**Pull request merged**